var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -87,17 +88,20 @@ def libs(self):
         libs += find_libraries(['libmpicxx', 'libmpifort'], lib_dir)
         libs += find_libraries('libmpi', release_lib_dir)
         libs += find_system_libraries(['libdl', 'librt', 'libpthread'])
+
+        # Find libfabric for libmpi.so
+        libfabric_rpath = join_path(self.component_path, 'libfabric', 'lib')
+        if '+external-libfabric' in self.spec:
+            libfabric_rpath = os.path.dirname(
+                [s for s in self.spec['libfabric'].libs if s.endswith('libfabric.so')][0]
+            )
+        libs += libfabric_rpath
```

Comment:
I agree, if that place exists it would make a better candidate. I have not been able to find a libs() function (or property) in the inheritance chain Libfabric<AutoToolsPackage<PackageBase. 

@rajachan Do you think it is worth adding libs() function to the libfabric package?

Reply:
Sure, go for it. If the oneapi packages are hitting this issue, others might too so it will be good to enumerate any custom paths via libs().

That said, all libs should already get installed inside the libfabric prefix's `lib` dir by default. 

```
$ tree install 
install
|-- bin
|   |-- fi_info
|   |-- fi_pingpong
|   `-- fi_strerror
|-- include
|   `-- rdma
|       |-- fabric.h
|       |-- fi_atomic.h
|       |-- fi_cm.h
|       |-- fi_collective.h
|       |-- fi_domain.h
|       |-- fi_endpoint.h
|       |-- fi_eq.h
|       |-- fi_errno.h
|       |-- fi_ext.h
|       |-- fi_rma.h
|       |-- fi_tagged.h
|       `-- fi_trigger.h
|-- lib
|   |-- libfabric.a
|   |-- libfabric.la
|   |-- libfabric.so -> libfabric.so.1.17.0
|   |-- libfabric.so.1 -> libfabric.so.1.17.0
|   |-- libfabric.so.1.17.0
|   `-- pkgconfig
|       `-- libfabric.pc

<snip>
```

I say that without reading the changes in the PR though, will take a closer look at this later today.

Reply:
So you are saying libs found in `/installation/root/lib` are added to the `libs()` function by default? That would make sense.

Reply:
> So you are saying libs found in /installation/root/lib are added to the libs() function by default? That would make sense.

If you're asking about `libfabric`, then yes: Because the library name `libfabric.so` matches the name of the package (`libfabric`), this will be handled automatically.

Reply:
OK, so I am using the full `self.spec['libfabric'].libs` now. No changes to the libfabric package necessary.